### DISCUSSION TITLE
Extract value for some element in the editor for reusing

### DISCUSSION THREAD

**Discussion by estebanpanelli** [react-page/react-page#1035] [Q&A]
Is there a way to obtain a component's state for reusing it in other places?

**Comment by macrozone** [comment]
hi @estebanpanelli ,

what is the exact use case?

You could do:

````
const YourContext = React.createContext()


const YourCellPlugin = {
 // ...
   Renderer: (props) => {
     const setValue = useContext(YourContext)
     useEffect(() => {
        setValue(props.data)
     }, [setValue, props.data]
     return ...
   }
}


// and somewhere

const [value, setValue] = useState()

<YourContext.Provider value={setValue}>
   // ...
   <Editor ... />
</YourContext.Provider>

```

**Reply by estebanpanelli** [reply]
Hi @macrozone, thanks for your reply.

I was thinking in visual components made in the editor (an entire row, a single column or a component with childs like background) with some complexity, could be saved it's value in a kind of library for reusing

**Reply by macrozone** [reply]
aaah, yes, i thought about implementing exactly this!

See 

https://github.com/react-page/react-page/issues/953

**Reply by estebanpanelli** [reply]
I'm thinking in a solution that keeps this functionality in the Editor layer requiring no coding.
Your proposed code may help, thanks!!
